pkg/daemon/cleanup/cleanup.go

```diff
@@ -33,30 +35,41 @@ import (
 	"github.com/openshift/windows-machine-config-operator/pkg/daemon/winsvc"
 	"github.com/openshift/windows-machine-config-operator/pkg/nodeconfig"
 	"github.com/openshift/windows-machine-config-operator/pkg/servicescm"
+	"github.com/openshift/windows-machine-config-operator/pkg/windows"
 )
 
 // Execute removes an instance from the cluster. If the parameter flag is true, the node object is not deleted.
 // If we are able to get the services ConfigMap tied to the desired version, all services defined in it are cleaned up.
-// TODO: Best effort cleanup if node object, desired version annotation, or services ConfigMap are not attainable
+// Otherwise, perform cleanup based on a combination of the OpenShift managed tag and the latest services ConfigMap.
 func Execute(c client.Client, addrs []net.Addr, svcMgr manager.Manager, preserveNode bool) error {
 	ctx := context.TODO()
 	node, err := controller.GetAssociatedNode(c, addrs)
 	if err != nil {
-		klog.Exitf("could not node object associated with this instance: %s", err.Error())
+		// If the node object is not attainable use a best effort cleanup stategy
+		klog.Warning(err.Error())
+		return outOfSyncCleanup(c, ctx, svcMgr, node, preserveNode)
 	}
 	klog.Infof("cleaning up managed services on node %s", node.Name)
 
 	desiredVersion, present := node.Annotations[nodeconfig.DesiredVersionAnnotation]
 	if !present {
-		return errors.Errorf("node missing desired version annotation")
+		// If the desired version annotation is not present, use a best effort cleanup stategy
+		klog.Warning("node missing desired version annotation")
+		return outOfSyncCleanup(c, ctx, svcMgr, node, preserveNode)
 	}
 	// Fetch the CM of the desired version
 	cm := &core.ConfigMap{}
 	err = c.Get(ctx,
 		client.ObjectKey{Namespace: controller.WmcoNamespace, Name: servicescm.NamePrefix + desiredVersion}, cm)
 	if err != nil {
+		if k8sapierrors.IsNotFound(err) {
+			klog.Warningf("unable to find desired version ConfigMap %s", servicescm.NamePrefix+desiredVersion)
+			// If the ConfigMap tied to the desired version annotation is not present, use a best effort cleanup stategy
+			return outOfSyncCleanup(c, ctx, svcMgr, node, preserveNode)
+		}
 		return err
 	}
+
 	return cleanCMServices(c, ctx, svcMgr, cm, node, preserveNode)
```

Comment:
Theres a lot of repeating syntax here with error checks + log warning + doing the outOfSyncCleanup.
This can be fixed by using an anonymous func, or a named func depending on what works best.
Take a look at the nodeconfig code to what [we're doing there](https://github.com/openshift/windows-machine-config-operator/blob/3492116e5071d2184226b67d47ce04040beac30d/pkg/nodeconfig/nodeconfig.go#L204).

This function can be refactored like this:
```
	cm := &core.ConfigMap{}
	err := func()error {
		node, err := controller.GetAssociatedNode(c, addrs)
		if err != nil {
			// If the node object is not attainable use a best effort cleanup stategy
			return err
		}

		desiredVersion, present := node.Annotations[nodeconfig.DesiredVersionAnnotation]
		if !present {
			// If the desired version annotation is not present, use a best effort cleanup stategy
			return errors.Wrap("node missing desired version annotation",err)
		}
		// Fetch the CM of the desired version
		err = c.Get(ctx,
			client.ObjectKey{Namespace: controller.WmcoNamespace, Name: servicescm.NamePrefix + desiredVersion}, cm)
		if err != nil {
			return errors.Wrap()
		}
		return nil
	}()
	if err == nil {
		// no issue getting cm
		// return with CM cleanup
	} 
	klog.Warningf(err)
	// return with out of sync cleanup
```